src/util.c

```diff
@@ -359,6 +359,18 @@ const char *get_environment(const char *key)
     return getenv(key);
 }
 
+#ifdef _WIN32
```

Comment:
This implementation of set_environment shouldn't pass the unit tests due to not implementing the `override` parameter. The `#ifdef _WIN32` needs to be removed as well. I didn't know that `putenv` was available on Windows, but since that's the case, I think the right fix is to have `get_environment` call `getenv` without doing anything else. Then `set_environment` should be:
```
if (override || getenv(key) == NULL) {
    char *str;
    int len = asprintf(&str, "%s=%s", key, value);
    if (len < 0)
        err(EXIT_FAILURE, "asprintf");
    putenv(str);
}
return 0;
```
If `asprintf` isn't on Windows, then I'll be sad. The way that you wrote it is ok, but delete the comment about Windows, since the reason is that `setenv` isn't available (i.e., the `#ifndef HAVE_SETENV`). I think that Coverity will complain about the return value from `malloc` not being defined, so could you add a check there, too.